### PR TITLE
Add details for mutable and structured objects

### PR DESCRIPTION
**Pull request by rphillips-cc**

### CODE REVIEW

File: articles/editing-objects-in-your-data.mdx [hunk 1]
```diff
@@ -1,97 +1,268 @@
...
+  - using-the-configuration-cascade
+  - array-structures-reference
+  - defining-what-adds-to-an-array-with-array-structures
+related_links:
+  - name: Material Icons
+    url: 'https://material.io/tools/icons/'
 ---
 
-Object inputs in your data show as a button in the data editor. This button opens a new editor to the side, showing the inputs inside that object. This continues as you click into nested Object inputs. Each object level has a back button to return to the parent scope.
+Object inputs represent object data within the Data Editor. They are ideal for grouping related inputs, and hiding complicated or infrequently edited data.
```
Comment (bglw) [inline articles/editing-objects-in-your-data.mdx:20]:
```suggestion
Object inputs represent object data within the Data Editor. They are ideal for grouping related inputs, or hiding complicated or infrequently edited data.
```

File: articles/editing-objects-in-your-data.mdx [hunk 2]
```diff
@@ -1,97 +1,440 @@
...
+call_to_action: contact-us
+
+_inputs:
+  $:
+    type: object
+    options:
+      place_groups_below: false
+      groups:
+        - heading: SEO
+          description: Values for search engines to read
```
Comment (rycoll) [inline articles/editing-objects-in-your-data.mdx]:
```suggestion
          comment: Values for search engines to read
```

File: articles/editing-objects-in-your-data.mdx [hunk 3]
```diff
@@ -1,97 +1,440 @@
...
+    Groups are labelled with a heading and description. They can start collapsed or expanded (default). The inputs that belong to each group are referred to by key. Groups can only contain inputs inside this object.
+
+    Ungrouped inputs are displayed after the groups by default, but you can reverse this with `place_groups_below`.
+
+    *Does not apply to mutable or structured objects*.
+  </comp.DataReferenceRow>
+  <comp.DataReferenceRow label="groups[*].heading" type_markdown="text">
+    The main text displayed at the top of a group. Visible when open or collapsed.
+  </comp.DataReferenceRow>
+  <comp.DataReferenceRow label="groups[*].description" type_markdown="text">
```
Comment (rycoll) [inline articles/editing-objects-in-your-data.mdx:424]:
```suggestion
  <comp.DataReferenceRow label="groups[*].comment" type_markdown="text">
```

**Review by rycoll — CHANGES_REQUESTED** [review]
I think `groups[*].description` should be `groups[*].comment`?

**Review by rycoll — APPROVED** [review]

File: articles/editing-objects-in-your-data.mdx (hunk 2 above)
Comment (rphillips-cc) [reply articles/editing-objects-in-your-data.mdx]: It's `description` here.

**Pull request merged**